### DISCUSSION TITLE
Feature Request - Add X-Frame-Options customizability

### DISCUSSION THREAD

**Discussion by styris-ame** [dani-garcia/vaultwarden#3929] [Ideas]
Since the closure of [issue 44](https://github.com/dani-garcia/vaultwarden/issues/44), it is not possible to embed vaultwarden as an iFrame on webapps like Homarr or organizr.

Adding an option to allow a specific website(s) to use iFrames would resolve this issue. Something like this should work:
`res.set_raw_header("X-Frame-Options", "ALLOW-FROM <WEBSITE>");`
An example for `<WEBSITE>` might be: `https://homarr.my-domain.com`

Perhaps a docker environment variable could be used to specify this.

**Comment by BlackDex** [comment]
We already have this, just not via that specific header but via CSP, which overrules the header you mentioned.

See: https://github.com/dani-garcia/vaultwarden/blob/bc26bfa589c007da9b9be37e1172060f38a948b9/.env.template#L342

Also see:https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors

**Comment by styris-ame** [comment]
Oh awesome, thank you!
